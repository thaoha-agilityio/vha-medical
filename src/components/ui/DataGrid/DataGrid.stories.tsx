// Libs
import type { Meta, StoryObj } from '@storybook/react';

// Types
import { ColumnType } from '@/types';

// Mocks
import { MOCK_APPOINTMENTS, MOCK_COLUMNS_APPOINTMENTS } from '@/mocks';

// Components
import DataGrid from '.';

const meta = {
  title: 'Components/DataGrid',
  tags: ['autodocs'],
  component: DataGrid,
  parameters: {
    layout: 'centered',
    nextjs: {
      appDirectory: true,
    },
  },
  argTypes: {
    data: { description: 'data of body table' },
    columns: { description: 'columns of table' },
    pagination: { description: 'pagination of table' },
  },
} as Meta<typeof DataGrid>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  decorators: [
    (Story) => (
      <div className="w-[900px]">
        <Story />
      </div>
    ),
  ],
  args: {
    data: MOCK_APPOINTMENTS,
    columns: MOCK_COLUMNS_APPOINTMENTS as ColumnType<unknown>[],
  },
};

export const HasDivider: Story = {
  decorators: [
    (Story) => (
      <div className="w-[900px]">
        <Story />
      </div>
    ),
  ],
  args: {
    data: MOCK_APPOINTMENTS,
    columns: MOCK_COLUMNS_APPOINTMENTS as ColumnType<unknown>[],
    hasDivider: true,
  },
};

export const Empty: Story = {
  decorators: [
    (Story) => (
      <div className="w-[900px]">
        <Story />
      </div>
    ),
  ],
  args: {
    data: [],
    columns: MOCK_COLUMNS_APPOINTMENTS as ColumnType<unknown>[],
  },
};
