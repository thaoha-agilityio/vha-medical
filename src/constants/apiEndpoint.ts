export const API_ENDPOINT = {
  AUTH: '/auth/local',
  APPOINTMENTS: '/appointments',
  NOTIFICATIONS: '/notifications',
  SPECIALTIES: '/specialties',
  USERS: '/users',
  UPLOAD: '/upload',
  CHEMISTS: '/chemists',
  PERMISSIONS: '/users-permissions',
};

export const API_URL = `${process.env.NEXT_PUBLIC_API_URL}/api`;
export const IMGBB_URL = `${process.env.NEXT_PUBLIC_IMGBB_URL}?key=${process.env.NEXT_PUBLIC_IMGBB_API_KEY}`;

export const AUTH_TOKEN = process.env.NEXT_PUBLIC_AUTH_TOKEN;
