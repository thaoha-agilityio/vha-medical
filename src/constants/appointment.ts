import { AppointmentModel, AppointmentStatusOption } from '@/types';
import { DEFAULT_CHEMIST_DATA } from './chemist';

export const APPOINTMENT_STATUS_OPTIONS: AppointmentStatusOption[] = [
  {
    key: 'new',
    label: 'New',
    value: 0,
  },
  {
    key: 'meeting',
    label: 'Meeting',
    value: 1,
  },
  {
    key: 'cancelled',
    label: 'Cancelled',
    value: 2,
  },
];

export const APPOINTMENT_STATUS = [
  {
    key: '0',
    label: 'New',
  },
  {
    key: '1',
    label: 'Meeting',
  },
  {
    key: '2',
    label: 'Canceled',
  },
];

export const DEFAULT_APPOINTMENT_DATA: AppointmentModel = {
  startTime: '',
  durationTime: '',
  status: 0,
  receiverId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
  senderId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
};

export const APPOINTMENT_SEARCH_PARAMS = ['receiverId', 'senderId'] as const;
