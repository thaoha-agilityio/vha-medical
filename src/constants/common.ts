export const RESULT_NOT_FOUND = 'Result Not Found';
