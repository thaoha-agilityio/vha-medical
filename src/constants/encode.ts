export const ENCODE_COOKIE = {
  algorithm: 'aes-256-cbc',
  secret: process.env.COOKIE_SECRET,
};
