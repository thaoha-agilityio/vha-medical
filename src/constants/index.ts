export * from './apiEndpoint';
export * from './appointment';
export * from './common';
export * from './firebase';
export * from './image';
export * from './message';
export * from './navigation';
export * from './pagination';
export * from './regex';
export * from './router';
export * from './timing';
export * from './chemist';
export * from './encode';
