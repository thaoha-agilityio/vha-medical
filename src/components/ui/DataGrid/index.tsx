'use client';

import { usePathname, useRouter, useSearchParams } from 'next/navigation';
import {
  Suspense,
  TransitionStartFunction,
  lazy,
  memo,
  useCallback,
  useMemo,
} from 'react';

// Constants
import { PAGE_DEFAULT, RESULT_NOT_FOUND } from '@/constants';

// Types
import { APIResponse, ColumnType, MetaResponse } from '@/types';

// Utils
import { cn, getObjectValue } from '@/utils';

// Components
import {
  Table,
  TableBody,
  TableCell,
  TableColumn,
  TableHeader,
  TableProps,
  TableRow,
} from '@nextui-org/react';
import { Spinner, Text } from '@/components/ui';

const Pagination = lazy(() => import('@/components/ui/Pagination'));

export interface DataTableProps<T> extends MetaResponse, TableProps {
  data: APIResponse<T>[];
  columns: ColumnType<T>[];
  hasDivider?: boolean;
  startTransition?: TransitionStartFunction;
  classWrapper?: string;
  classRow?: string;
  classCell?: string;
}

const DataGrid = memo(
  <T,>({
    data,
    columns,
    pagination,
    hasDivider = false,
    classWrapper = '',
    classRow = '',
    classCell = '',
    startTransition,
    ...props
  }: DataTableProps<T>) => {
    const { page = PAGE_DEFAULT, pageCount = PAGE_DEFAULT } = pagination ?? {};
    const searchParams = useSearchParams() ?? '';
    const pathname = usePathname() ?? '';
    const { replace } = useRouter();
    const params = useMemo(
      () => new URLSearchParams(searchParams),
      [searchParams],
    );

    const handleReplaceURL = useCallback(
      (params: URLSearchParams) => {
        startTransition?.(() => {
          replace(`${pathname}?${params.toString()}`);
        });
      },
      [pathname, replace, startTransition],
    );

    const handlePageChange = useCallback(
      (page: number) => {
        if (page === 1) {
          params.delete('page');
        } else {
          params.set('page', `${page}`);
        }

        handleReplaceURL(params);
      },
      [handleReplaceURL, params],
    );

    const classDivider =
      'border-0 border-primary-100 border-b border-opacity-10';

    return (
      <>
        <Table
          hideHeader
          className="w-full"
          tabIndex={0}
          classNames={{
            emptyWrapper: 'text-primary-100 text-xl font-medium h-[152px] pb-4',
            wrapper: cn(
              `bg-transparent-200 shadow-none p-0 ${classWrapper ?? ''}`,
            ),
          }}
          aria-label="Table"
          {...props}
        >
          <TableHeader>
            {columns.map((column) => {
              return <TableColumn key={column.key}>{column.title}</TableColumn>;
            })}
          </TableHeader>
          <TableBody
            emptyContent={RESULT_NOT_FOUND}
            loadingContent={<Spinner size="lg" className="absolute" />}
          >
            {data.length
              ? data.map((item, index) => {
                  const id = getObjectValue(item, 'id');
                  const isLastItem = data.length === index + 1;
                  return (
                    <TableRow
                      key={id}
                      className={!isLastItem ? classRow ?? '' : ''}
                    >
                      {columns.map((column) => (
                        <TableCell
                          key={`table-row-cell-${column.key}`}
                          className={cn(
                            `p-0 ${!isLastItem ? classCell ?? '' : ''}`,
                            `${hasDivider ? (!isLastItem ? `py-3 ${classDivider}` : 'pt-3') : ''}`,
                            column.additionalClassName,
                          )}
                          role="cell"
                          headers={`${props.id}-${column.key}`}
                        >
                          {column.customNode ? (
                            column.key === 'actions' ? (
                              column.customNode({ item: item.attributes, id })
                            ) : (
                              column.customNode({
                                column,
                                item: item.attributes,
                              })
                            )
                          ) : (
                            <Text variant="error" size="xs">
                              {getObjectValue(item.attributes, column.key)}
                            </Text>
                          )}
                        </TableCell>
                      ))}
                    </TableRow>
                  );
                })
              : []}
          </TableBody>
        </Table>
        {!!pagination && pagination.pageCount > 1 && (
          <Suspense fallback={null}>
            <Pagination
              classNames={{ base: 'mt-4' }}
              initialPage={page}
              total={pageCount}
              onChange={handlePageChange}
            />
          </Suspense>
        )}
      </>
    );
  },
);

DataGrid.displayName = 'DataGrid';
export default DataGrid;
