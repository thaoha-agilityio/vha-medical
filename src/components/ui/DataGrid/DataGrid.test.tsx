import { RenderResult, act, render } from '@testing-library/react';

// Mocks
import {
  MOCK_APPOINTMENTS,
  MOCK_COLUMNS_APPOINTMENTS,
  MOCK_PAGINATION,
} from '@/mocks';

// Components
import DataGrid, { DataTableProps } from '.';
import { RESULT_NOT_FOUND } from '@/constants';

const mockReplace = jest.fn();

// Mock next
jest.mock('next/navigation', () => ({
  useSearchParams: jest.fn(() => {
    const params = new URLSearchParams();
    return { params };
  }),
  usePathname: jest.fn(),
  useRouter: () => ({ replace: mockReplace }),
}));

jest.mock('@/utils', () => ({
  cn: jest.fn(() => 'mocked-class'),
  getObjectValue: jest.fn(),
  formatTimeAppointment: jest.fn(),
  formatNewDate: jest.fn(),
}));

describe('DataGrid', () => {
  const mockProps = {
    data: MOCK_APPOINTMENTS,
    columns: MOCK_COLUMNS_APPOINTMENTS,
    pagination: MOCK_PAGINATION,
  } as DataTableProps<unknown>;

  let renderResult: RenderResult;
  const dataGridComponent = <DataGrid {...mockProps} />;

  beforeEach(async () => {
    renderResult = await act(() => render(dataGridComponent));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Should matches snapshot', async () => {
    jest.spyOn(URLSearchParams.prototype, 'get').mockReturnValue('1');

    const { container } = renderResult;

    expect(container).toMatchSnapshot();
  });

  it('Should show empty result', async () => {
    jest.spyOn(URLSearchParams.prototype, 'get').mockReturnValue('1');

    const { getByText } = await act(() =>
      render(<DataGrid {...mockProps} data={[]} />),
    );

    expect(getByText(RESULT_NOT_FOUND)).toBeInTheDocument();
  });
});
