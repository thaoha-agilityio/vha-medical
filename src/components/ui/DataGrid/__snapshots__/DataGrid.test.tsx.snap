// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrid Should matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col relative gap-4 w-full"
  >
    <div
      class="p-4 z-0 flex flex-col relative justify-between gap-4 bg-content1 overflow-auto rounded-large shadow-small w-full mocked-class"
    >
      <table
        aria-describedby=""
        aria-label="Table"
        class="min-w-full h-auto table-auto w-full"
        id="react-aria-:r0:"
        role="grid"
        tabindex="0"
      >
        <thead
          class="[&>tr]:first:rounded-lg hidden"
          role="rowgroup"
        >
          <tr
            class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
            role="row"
          >
            <th
              class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
              data-key="senderId"
              id="react-aria-:r0:-senderId"
              role="columnheader"
              tabindex="-1"
            >
              Sender
            </th>
            <th
              class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
              data-key="receiverId"
              id="react-aria-:r0:-receiverId"
              role="columnheader"
              tabindex="-1"
            >
              Receiver
            </th>
            <th
              class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
              data-key="durationTime"
              id="react-aria-:r0:-durationTime"
              role="columnheader"
              tabindex="-1"
            >
              Duration
            </th>
            <th
              class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
              data-key="startTime"
              id="react-aria-:r0:-startTime"
              role="columnheader"
              tabindex="-1"
            >
              Date
            </th>
            <th
              class="group px-3 h-10 align-middle bg-default-100 whitespace-nowrap text-foreground-500 text-tiny font-semibold first:rounded-s-lg last:rounded-e-lg data-[sortable=true]:cursor-pointer data-[hover=true]:text-foreground-400 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-start"
              data-key="status"
              id="react-aria-:r0:-status"
              role="columnheader"
              tabindex="-1"
            >
              Status
            </th>
          </tr>
          <tr
            aria-hidden="true"
            class="w-px h-px block"
            style="margin-left: 0.25rem; margin-top: 0.25rem;"
            tabindex="-1"
          />
        </thead>
        <tbody
          role="rowgroup"
        >
          <tr
            aria-labelledby="react-aria-:r0:-$.5.0-senderId"
            class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
            data-first="true"
            data-key="$.5.0"
            role="row"
            tabindex="-1"
          >
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-senderId"
              headers="undefined-senderId"
              id="react-aria-:r0:-$.5.0-senderId"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              >
                Alex Stanton
              </p>
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-receiverId"
              headers="undefined-receiverId"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              >
                Jessica Jane
              </p>
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-durationTime"
              headers="undefined-durationTime"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              />
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-startTime"
              headers="undefined-startTime"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              />
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-status"
              headers="undefined-status"
              role="cell"
              tabindex="-1"
            >
              <div
                class="mocked-class"
              >
                <p
                  class="text-2xs font-semibold capitalize"
                >
                  new
                </p>
              </div>
            </td>
          </tr>
          <tr
            aria-labelledby="react-aria-:r0:-$.5.1-senderId"
            class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
            data-key="$.5.1"
            data-middle="true"
            data-odd="true"
            role="row"
            tabindex="-1"
          >
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-senderId"
              headers="undefined-senderId"
              id="react-aria-:r0:-$.5.1-senderId"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              >
                Alex Stanton
              </p>
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-receiverId"
              headers="undefined-receiverId"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              >
                ThangHq
              </p>
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-durationTime"
              headers="undefined-durationTime"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              />
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-startTime"
              headers="undefined-startTime"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              />
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-status"
              headers="undefined-status"
              role="cell"
              tabindex="-1"
            >
              <div
                class="mocked-class"
              >
                <p
                  class="text-2xs font-semibold capitalize"
                >
                  new
                </p>
              </div>
            </td>
          </tr>
          <tr
            aria-labelledby="react-aria-:r0:-$.5.2-senderId"
            class="group outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
            data-key="$.5.2"
            data-last="true"
            role="row"
            tabindex="-1"
          >
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-senderId"
              headers="undefined-senderId"
              id="react-aria-:r0:-$.5.2-senderId"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              >
                ThangHq
              </p>
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-receiverId"
              headers="undefined-receiverId"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              >
                Alex Stanton
              </p>
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-durationTime"
              headers="undefined-durationTime"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              />
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-startTime"
              headers="undefined-startTime"
              role="cell"
              tabindex="-1"
            >
              <p
                class="mocked-class"
              />
            </td>
            <td
              class="py-2 px-3 relative align-middle whitespace-normal text-small font-normal [&>*]:z-1 [&>*]:relative outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 before:content-[''] before:absolute before:z-0 before:inset-0 before:opacity-0 data-[selected=true]:before:opacity-100 group-data-[disabled=true]:text-foreground-300 group-data-[disabled=true]:cursor-not-allowed before:bg-default/60 data-[selected=true]:text-default-foreground first:before:rounded-s-lg last:before:rounded-e-lg text-start mocked-class"
              data-key="table-row-cell-status"
              headers="undefined-status"
              role="cell"
              tabindex="-1"
            >
              <div
                class="mocked-class"
              >
                <p
                  class="text-2xs font-semibold capitalize"
                >
                  meeting
                </p>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <nav
    aria-label="pagination navigation"
    class="p-2.5 -m-2.5 overflow-x-scroll scrollbar-hide mt-4"
    data-active-page="1"
    data-controls="true"
    data-dots-jump="5"
    data-slot="base"
    data-total="3"
    role="navigation"
  >
    <ul
      class="flex flex-nowrap h-fit max-w-fit relative gap-1 items-center overflow-visible rounded-medium"
      data-slot="wrapper"
    >
      <span
        aria-hidden="true"
        class="absolute overflow-visible items-center justify-center origin-center left-0 select-none touch-none pointer-events-none z-20 bg-primary text-primary-foreground data-[moving=true]:transition-transform !data-[moving=true]:duration-300 opacity-0 data-[moving]:opacity-100 min-w-9 w-9 h-9 text-small rounded-medium hidden"
        data-moving="false"
        data-slot="cursor"
        style="transform: translateX(0px) scale(1);"
      >
        1
      </span>
      <li
        aria-disabled="true"
        aria-label="previous page button"
        class="flex flex-wrap truncate box-border items-center justify-center text-default-foreground outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium mocked-class"
        data-disabled="true"
        data-slot="prev"
        tabindex="-1"
      >
        <svg
          aria-hidden="true"
          fill="none"
          focusable="false"
          height="1em"
          role="presentation"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M15.5 19l-7-7 7-7"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </li>
      <li
        aria-current="true"
        aria-label="pagination item 1 active"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center text-default-foreground outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium mocked-class"
        data-active="true"
        data-slot="item"
        tabindex="0"
      >
        1
      </li>
      <li
        aria-label="pagination item 2"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center text-default-foreground outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium mocked-class"
        data-slot="item"
        tabindex="0"
      >
        2
      </li>
      <li
        aria-label="pagination item 3"
        class="tap-highlight-transparent select-none touch-none data-[pressed=true]:scale-[0.97] transition-transform-background flex flex-wrap truncate box-border items-center justify-center text-default-foreground outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium mocked-class"
        data-slot="item"
        tabindex="0"
      >
        3
      </li>
      <li
        aria-label="next page button"
        class="flex flex-wrap truncate box-border items-center justify-center text-default-foreground outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[disabled=true]:text-default-300 data-[disabled=true]:pointer-events-none shadow-sm bg-default-100 [&[data-hover=true]:not([data-active=true])]:bg-default-200 active:bg-default-300 min-w-9 w-9 h-9 text-small rounded-medium mocked-class"
        data-slot="next"
        tabindex="0"
      >
        <svg
          aria-hidden="true"
          class="rotate-180"
          fill="none"
          focusable="false"
          height="1em"
          role="presentation"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M15.5 19l-7-7 7-7"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </li>
    </ul>
  </nav>
</div>
`;
